edfio/edf.py

```diff
@@ -276,6 +278,15 @@ def data(self) -> npt.NDArray[np.float64]:
         data.setflags(write=False)
         return data
 
+    @data.setter
+    def data(self, data: npt.NDArray[np.float64]) -> None:
+        if len(data) != len(self._digital):
+            raise ValueError(
+                f"Signal lengths must match: got {len(data)}, expected {len(self._digital)}."
+            )
+        self._set_physical_range(None, data)
```

Comment:
Should we log a warning if calibration data is changed? This might not be wanted/expected behavior. What if the user wants to keep the calibration data as is? can we add an `edf.replaceSignal(...)` method such that the user can update a signal and set calibration data explicitly?

Reply:
As discussed in our call, 8e5f1ed replaces the setter with a method `update_data`. If the parameter `keep_physical_range=False` (introduced in e216fb6f) is set to `True`, the calibration is not changed and a `ValueError` is raised if the new data exceeds the existing `physical_range`.